lib/src/server/server.dart

```diff
@@ -101,20 +102,35 @@ class Server {
     server.listen((socket) {
       final connection = new ServerTransportConnection.viaSocket(socket);
       _connections.add(connection);
+      ServerHandler_ handler;
       // TODO(jakobr): Set active state handlers, close connection after idle
       // timeout.
-      connection.incomingStreams.listen(serveStream, onError: (error) {
+      connection.incomingStreams.listen((stream) {
+        handler = serveStream_(stream);
+      }, onError: (error) {
         print('Connection error: $error');
       }, onDone: () {
+        // TODO(sigurdm): This is not correct behavior in the presence of
+        // half-open tcp streams.
```

Comment:
I think it's better to call them 'half-closed', as half-open is something else... sorry, I might have used the wrong terminology from time to time :)